lib/plugins/aws/package/compile/events/s3/index.js

```diff
@@ -10,7 +10,7 @@ class AwsCompileS3Events {
     this.options = options;
     this.provider = this.serverless.getProvider('aws');
 
-    this.allowedBucketProperties = _.keyBy([
+    this.allowedBucketProperties = [
```

Comment:
Let's use `Set`, and update the logic that relies on it, to work with _set_.

It should turn more to the point, and simpler